ui/job-view/details/PinBoard.jsx

```diff
@@ -212,9 +194,21 @@ export default class PinBoard extends React.Component {
     return this.props.isLoggedIn && cancellableJobs.length > 0;
   }
 
-  async cancelAllPinnedJobs() {
+  cancelAllPinnedJobs() {
     if (window.confirm('This will cancel all the selected jobs. Are you sure?')) {
-      await JobModel.cancel(this.$rootScope.repoName, Object.keys(this.props.pinnedJobs));
+      const jobIds = Object.keys(this.props.pinnedJobs);
+
+      this.thNotify.send(
+        'Request sent to cancel all pinned jobs via actions.json',
```

Comment:
I think the UX might be confusing here (compared to the previous implementation), since the user might see "Request sent ..." (with a success colour) only to have the request actually fail, since the request doesn't actually occur until after the notification at the moment.

For these batch-jobs-action methods (`cancelAllPinnedJobs()`, `retriggerAllPinnedJobs()`, `cancelAllJobs()`), we presumably:
1. don't want to spam the user with a notification for each job (hence the new `notifyAfterEachRetrigger` added by this PR)
2. want some notification before any requests have been made, to let them know work is occurring - however this should probably be changed to use `thNotify` severity `'info'` rather than `'success'`
3. want some notification at the end to say all requests were successful (with `thNotify` severity `'success'`)

It seems like it would be easier to move this handling inside the new `JobModel.retrigger()` and `JobModel.cancel()` implementations added in this PR, since it would reduce duplication and at the moment the caller of them doesn't know if they throw (which is needed for (3) above). Then `notifyAfterEachRetrigger` could be removed, and replaced with a length check on `jobIDs` (ie if more than N jobs, don't show all messages). 

Thoughts?

Reply:
Agreed, that's definitely neater.